src/wallet/wallet.cpp

```diff
@@ -3663,6 +3677,7 @@ int CMerkleTx::SetMerkleBranch(const CBlock& block)
 
 int CMerkleTx::GetDepthInMainChain(bool enableIX) const
 {
+    AssertLockHeld(cs_main);
```

Comment:
I don't see any value on this AssertLockHeld, just a duplicated check. it's already being checked inside the `GetDepthInMainChain(pindexRet, enableIX)` that it's two lines below.